src/applications/ivc-champva/10-7959C/chapters/healthInsuranceInformation.js

```diff
@@ -11,43 +11,72 @@ import { applicantListSchema } from '../config/constants';
 import { applicantWording } from '../../shared/utilities';
 import ApplicantField from '../../shared/components/applicantLists/ApplicantField';
 
-export const applicantHasPrimarySchema = {
-  uiSchema: {
-    applicants: { items: {} },
-  },
-  schema: applicantListSchema([], {
-    applicantHasPrimary: {
-      type: 'object',
-      properties: {
-        hasPrimary: { type: 'string' },
-        _unused: { type: 'string' },
-      },
-    },
-  }),
+const MEDIGAP = {
+  medigapPlanA: 'Medigap Plan A',
+  medigapPlanB: 'Medigap Plan B',
+  medigapPlanC: 'Medigap Plan C',
+  medigapPlanD: 'Medigap Plan D',
+  medigapPlanF: 'Medigap Plan F',
+  medigapPlanG: 'Medigap Plan G',
+  medigapPlanK: 'Medigap Plan K',
+  medigapPlanL: 'Medigap Plan L',
+  medigapPlanM: 'Medigap Plan M',
 };
 
-export const applicantPrimaryProviderSchema = {
-  uiSchema: {
-    applicants: {
-      'ui:options': {
-        viewField: ApplicantField,
+/*
+Primary health insurance and secondary health insurance information use
+the same set of questions. This schema works for either depending on
+the boolean passed in (if true, we generate the primary schema, if false
+we generate the secondary schema). Using this pattern for all primary/secondary
+schemas
+*/
+export const applicantHasInsuranceSchema = isPrimary => {
+  const val = isPrimary ? 'Primary' : 'Secondary';
+  const keyname = `applicantHas${val}`;
+  const property = `has${val}`;
+  return {
+    uiSchema: {
+      applicants: { items: {} },
+    },
+    schema: applicantListSchema([], {
+      [keyname]: {
+        type: 'object',
+        properties: {
+          [property]: { type: 'string' },
+          _unused: { type: 'string' },
+        },
       },
-      items: {
-        ...titleUI(
-          ({ formData }) =>
-            `${applicantWording(formData)} health insurance provider’s name`,
-        ),
-        applicantPrimaryProvider: {
-          'ui:title': 'Provider’s name',
-          'ui:webComponentField': VaTextInputField,
+    }),
+  };
+};
+
+export const applicantProviderSchema = isPrimary => {
+  const keyname = `applicant${isPrimary ? 'Primary' : 'Secondary'}Provider`;
+  return {
+    uiSchema: {
+      applicants: {
+        'ui:options': {
+          viewField: ApplicantField,
+        },
+        items: {
+          ...titleUI(
+            ({ formData }) =>
+              `${applicantWording(formData)} ${
+                isPrimary ? '' : 'secondary'
+              } health insurance provider’s name`,
+          ),
+          [keyname]: {
+            'ui:title': 'Provider’s name',
+            'ui:webComponentField': VaTextInputField,
+          },
         },
       },
     },
-  },
-  schema: applicantListSchema(['applicantPrimaryProvider'], {
-    titleSchema,
-    applicantPrimaryProvider: { type: 'string' },
-  }),
+    schema: applicantListSchema([keyname], {
+      titleSchema,
+      [keyname]: { type: 'string' },
+    }),
+  };
```

Comment:
Adjusted these schemas to work for either primary or secondary health insurance screens. Plan is to reuse all of the primary health insurance schemas in this way for the secondary health insurance screens.